From 2ca0fff0401bba3f7590d66c472f79790ce125c6 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Tue, 28 Jul 2015 12:02:32 +0530
Subject: [PATCH] Wcove-gpio: Added changes to support power roles swap.

As per typec power role swap event changes in detect driver,
the extcon cable change listener is changed from USB-Host
to USB_TYPEC_SRC.

Change-Id: I66f39ca52688c1b61a2bde702632a4033e46dafd
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-14786
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/misc/intel_wcove_gpio.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/intel_wcove_gpio.c b/drivers/misc/intel_wcove_gpio.c
index 6ab7e9b9e407..c0cb44cfd877 100644
--- a/drivers/misc/intel_wcove_gpio.c
+++ b/drivers/misc/intel_wcove_gpio.c
@@ -66,7 +66,7 @@ static inline struct power_supply *wcove_gpio_get_psy_charger(void)
 
 	class_dev_iter_init(&iter, power_supply_class, NULL, NULL);
 	while ((dev = class_dev_iter_next(&iter))) {
-		pst = (struct power_supply *)dev_get_drvdata(dev);
+		pst = dev_get_drvdata(dev);
 		if (IS_CHARGER(pst)) {
 			class_dev_iter_exit(&iter);
 			return pst;
@@ -140,7 +140,7 @@ static int wcgpio_check_events(struct wcove_gpio_info *info,
 		return -ENOMEM;
 	}
 
-	evt->is_src_connected = extcon_get_cable_state(edev, "USB-Host");
+	evt->is_src_connected = extcon_get_cable_state(edev, "USB_TYPEC_SRC");
 	dev_info(&info->pdev->dev,
 			"[extcon notification] evt: Provider - %s\n",
 			evt->is_src_connected ? "Connected" : "Disconnected");
@@ -201,7 +201,7 @@ static int wcove_gpio_probe(struct platform_device *pdev)
 
 	info->nb.notifier_call = wcgpio_event_handler;
 	ret = extcon_register_interest(&info->otg_cable_obj, NULL,
-						"USB-Host",
+						"USB_TYPEC_SRC",
 						&info->nb);
 	if (ret) {
 		dev_err(&pdev->dev,
-- 
1.9.1

